# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../dietlibc/pkg_patch/pkg_iptables.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
diff -ruN iptables-1.3.1.orig/Makefile iptables-1.3.1/Makefile
--- iptables-1.3.1.orig/Makefile	2005-03-07 15:02:23.000000000 +0100
+++ iptables-1.3.1/Makefile	2005-07-14 03:35:37.000000000 +0200
@@ -103,7 +103,7 @@
 STATIC_LIBS  = extensions/libext.a
 STATIC6_LIBS = extensions/libext6.a
 LDFLAGS      = -static
-LDLIBS       =
+LDLIBS       = -lrpc 
 endif
 
 .PHONY: default
diff -ruN iptables-1.3.1.orig/include/libiptc/ipt_kernel_headers.h iptables-1.3.1/include/libiptc/ipt_kernel_headers.h
--- iptables-1.3.1.orig/include/libiptc/ipt_kernel_headers.h	2004-10-10 11:56:23.000000000 +0200
+++ iptables-1.3.1/include/libiptc/ipt_kernel_headers.h	2005-07-14 03:26:55.000000000 +0200
@@ -5,7 +5,7 @@
 
 #include <limits.h>
 
-#if defined(__GLIBC__) && __GLIBC__ == 2
+#if (defined(__GLIBC__) && __GLIBC__ == 2) || defined(__dietlibc__) 
 #include <netinet/ip.h>
 #include <netinet/in.h>
 #include <netinet/ip_icmp.h>
diff -ruN iptables-1.3.1.orig/iptables-restore.c iptables-1.3.1/iptables-restore.c
--- iptables-1.3.1.orig/iptables-restore.c	2005-01-04 11:38:39.000000000 +0100
+++ iptables-1.3.1/iptables-restore.c	2005-07-14 03:30:28.000000000 +0200
@@ -8,7 +8,7 @@
  */
 
 #include <getopt.h>
-#include <sys/errno.h>
+#include <errno.h>
 #include <string.h>
 #include <stdio.h>
 #include <stdlib.h>
diff -ruN iptables-1.3.1.orig/iptables-save.c iptables-1.3.1/iptables-save.c
--- iptables-1.3.1.orig/iptables-save.c	2005-01-04 11:38:38.000000000 +0100
+++ iptables-1.3.1/iptables-save.c	2005-07-14 03:31:53.000000000 +0200
@@ -6,7 +6,7 @@
  *
  */
 #include <getopt.h>
-#include <sys/errno.h>
+#include <errno.h>
 #include <stdio.h>
 #include <fcntl.h>
 #include <stdlib.h>
